src/mds/MDCache.cc

```diff
@@ -5310,6 +5310,8 @@ bool MDCache::process_imported_caps()
     cap_imports_num_opening++;
     dout(10) << "  opening missing ino " << p->first << dendl;
     open_ino(p->first, (int64_t)-1, new C_MDC_RejoinOpenInoFinish(this, p->first), false);
+    
+    mds->heartbeat_reset();  
```

Comment:
calling heartbeat reset for open_ino seems expensive. how about

if (!(cap_imports_num_opening % 1000))
 mds->heartbeat_reset();

Reply:
agree

Reply:
a second thought about this, cap_imports_num_opening can not represent the loop iteration times, it can only represent the concurrent open_ino call.
so, maybe we should introduce an iteration variable ?

Reply:
sorry, i was wrong, cap_imports_num_opening is right.